cfme/tests/platform/test_advanced_config.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+def set_config(appliance):
+    vmdb_config = appliance.advanced_settings
+    assert vmdb_config['http_proxy']['default']['host'] is None
+
+    # Check that the default state is None
+    vmdb_config['http_proxy']['default'] = {'host': 'bar'}
+    appliance.update_advanced_settings(vmdb_config)
+    vmdb_config = appliance.advanced_settings
+    assert vmdb_config['http_proxy']['default']['host'] == 'bar'
+    return vmdb_config
+
+
+@pytest.mark.ignore_stream("5.9")
+def test_advanced_config_reset_nonleaf(appliance):
+    '''Check whether we can use "<<reset>>" string to reset non-leaf element
+    of the advanced config.'''
+
+    vmdb_config = set_config(appliance)
+    vmdb_config['http_proxy']['default'] = '<<reset>>'
+    appliance.update_advanced_settings(vmdb_config)
+    vmdb_config = appliance.advanced_settings
+    assert vmdb_config['http_proxy']['default']['host'] is None
+
+
+@pytest.mark.ignore_stream("5.9")
+def test_advanced_config_reset_leaf(appliance):
+    '''Check whether we can use "<<reset>>" string to reset the leaf element
```

Comment:
I think this should just be parametrized and a single test function.

By parametrizing we can also define the config leaf in a more programmatic way instead of repeating the dictionary path 7 times.

Reply:
Done